### ISSUE TITLE
Can't change multicast TTL of an IPv6 socket on Linux

### ISSUE THREAD

**Issue by PawelGerr** [dotnet/corefx#29781]
I'm getting an error when trying to change the multicast TTL of an IPv6 socket on ubuntu 16.04. There are no errors when having an IPv4 socket or running on windows 10.

dotnet version: 2.1.101
OS: ubuntu 16.04 x64

Error:
```
System.Net.Sockets.SocketException: Operation not supported
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
   at System.Net.Sockets.Socket.SetSocketOption(SocketOptionLevel optionLevel, SocketOptionName optionName, Int32 optionValue, Boolean silent)
   at System.Net.Sockets.Socket.SetSocketOption(SocketOptionLevel optionLevel, SocketOptionName optionName, Int32 optionValue)
   at MulticastTtl.Program.Main(String[] args) in /mnt/hgfs/Tests/MulticastTtl/MulticastTtl/Program.cs:line 15
```

Repro: https://github.com/PawelGerr/DotnetCoreUdpMulticastTtl/tree/master/MulticastTtl
Executable: https://github.com/PawelGerr/DotnetCoreUdpMulticastTtl/tree/master/MulticastTtl/publish

.csproj-File
```xml
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>netcoreapp2.0</TargetFramework>
    </PropertyGroup>
</Project>
```

Program.cs
```c#
using System;
using System.Net.Sockets;

namespace MulticastTtl
{
	class Program
	{
		static void Main(string[] args)
		{
			// var addressFamily = AddressFamily.InterNetwork;
			var addressFamily = AddressFamily.InterNetworkV6;

			var socket = new Socket(addressFamily, SocketType.Dgram, ProtocolType.Udp);

			socket.SetSocketOption(addressFamily == AddressFamily.InterNetwork 
				? SocketOptionLevel.IP 
				: SocketOptionLevel.IPv6, SocketOptionName.MulticastTimeToLive, 32);
		}
	}
}
```
[EDIT] Add syntax highlighting by @karelz

**Comment by karelz** [comment]
@wfurt any idea?
cc @geoffkizer

**Comment by wfurt** [comment]
I'll give it try @karelz This may be similar to fixes in #25544

According to man page:

IP_MULTICAST_TTL | Set Time To Live (TTL) in the IP header for outgoing multicast datagrams. An option_value parameter of type char is used to set this value between 0 and 255. This option is only supported for sockets with an address family of AF_INET and type of SOCK_DGRAM or SOCK_RAW.

What we should probably use instead is:

IPV6_MULTICAST_HOPS | Set the hop limit value that will be used for subsequent multicast packets sent by this socket. An option_value parameter of type int is used to set this value between 0 and 255. If IPV6_MULTICAST_HOPS is not set, the default is 1.   This option is only supported for sockets with an address family of AF_INET6 and type of SOCK_DGRAM or SOCK_RAW

This is probably valid bug and fix should be probably hidden in Unix PAL layer.
